pkg/component/nodemanagement/machinecontrollermanager/machine_controller_manager.go

```diff
@@ -119,21 +122,51 @@ func (m *machineControllerManager) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	if _, err := controllerutils.GetAndCreateOrStrategicMergePatch(ctx, m.client, clusterRoleBinding, func() error {
-		clusterRoleBinding.OwnerReferences = []metav1.OwnerReference{{
-			APIVersion:         "v1",
-			Kind:               "Namespace",
-			Name:               m.namespace,
-			UID:                m.values.namespaceUID,
-			Controller:         ptr.To(true),
-			BlockOwnerDeletion: ptr.To(true),
-		}}
-		clusterRoleBinding.RoleRef = rbacv1.RoleRef{
+	if _, err := controllerutils.GetAndCreateOrStrategicMergePatch(ctx, m.client, role, func() error {
```

Comment:
Do we really need to create a `Role` in each shoot namespace? Alternatively, we could also keep using the `ClusterRole`, and just create a `RoleBinding` that references it. This should also scope the MCMs permissions to only the namespaces the `RoleBinding` was created in. Thoughts?

Reply:
I did think of using a `ClusterRole/RoleBinding`, but decided to go with `Role/RoleBinding` as from a security pov it would be better accepted as the `ClusterRole` then can be misused by some other rogue actor 
I do not have a hard opinion on this though

Reply:
> as from a security pov it would be better accepted as the `ClusterRole` then can be misused by some other rogue actor

Can you elaborate on this? How would another rogue actor misue the `ClusterRole`? Which scenario do you have in mind?

Reply:
I assume another pod can (accidently?) use this `ClusterRole` to get permissions to resources across all namespaces 
I get that it's a slightly weak argument, but it is the principle of least privilege

Reply:
I don't buy this (it seems rather artificial), but that's fine 😄 I just wanted to hear your thoughts. I'm fine with going forward with the `Role` (this also makes the code easier since https://github.com/gardener/gardener/blob/master/pkg/component/nodemanagement/machinecontrollermanager/bootstrap.go can be dropped entirely. Thank you!

Reply:
Done
PTAL